src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -766,6 +773,7 @@ public CompletableFuture<Void> setInstance(@NotNull Instance instance, @NotNull
         this.lastSyncedPosition = spawnPosition;
         this.previousPhysicsResult = null;
         this.instance = instance;
+        initCollisions();
```

Comment:
Could this be called in the constructor instead?

Reply:
The only way to set `entityType` is through the constructor and `switchEntityType`, and the field is marked with `UNSAFE to change, modify at your own risk`.

It's safe to call `updateCollisions` in the constructor after the field `entityType` is set and updating collisions in the constructor makes a lot more sense than in `setInstance`